addons/sale_margin/models/sale_order.py

```diff
@@ -2,11 +2,18 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo import api, fields, models
+from odoo.tools.sql import column_exists
 
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
+    def _auto_init(self):
+        if not column_exists(self.env.cr, "sale_order_line", "margin"):
+            # By creating the column 'margin' manually we steer clear of hefty data computation.
+            self.env.cr.execute("ALTER TABLE sale_order_line ADD COLUMN margin VARCHAR DEFAULT 'exception'")
```

Comment:
This is just plain wrong ...
* `margin` field column isn't supposed to be VARCHAR
* wtf is this DEFAULT 'exception' ?

Did you even look at the resulting database?
![image](https://user-images.githubusercontent.com/19364378/190992020-fa412ad5-3df9-4b39-82e8-f77d29649ac9.png)

The orm finds the wrong column and renames it, leaving a dumb empty and useless column in database.

Reply:
@Feyensv my bad for the column type, I corrected it.
we basically add the field with a default value, so that was the purpose of 'exception' for a numeric field I put 0
but what it does actually => the data will be gradually initialized as the rows get modified cs PG won't have to interfere with them.
apart from this, is there anything else that you see that might need to be improvised?
Thanks!